### PR TITLE
Swan v3.0

### PR DESCRIPTION
**Pull request by geoperez**
BREAKING CHANGES:

- Add: new IPropertyProxy from Mario
- Remove: IComponentCollection interface and children
- Remove: Winservices support classes
- Remove: PropertyDisplayAttribute and related methods

### CODE REVIEW

**Comment by rdeago** [comment]
`IComponentCollection` is used in EmbedIO. On the other hand, it is probably _only_ used in EmbedIO, so it belongs there, not in SWAN.

While you're at it: `IDataDictionary`, `DataDictionary`, and `ConcurrentDataDictionary` must die, or at least be obsoleted. I originally wrote them to avoid breaking code that relied on a certain behavior in EmbedIO's session data; as it turns out, though, that behavior was wrong, relying on it was just shameful defensive programming, and writing those classes was an experiment in futility. When compiled with nullable reference types, they trigger all kinds of warnings about their nullability semantics differing from the interfaces they implement.

**Comment by geoperez** [comment]
I can keep `IComponentCollection`, I was confused about this is really used in EmbedIO or not.

Thanks for the clarification.

**Comment by rdeago** [comment]
The real question IMHO would be, is anyone else using it? It's not so general-purpose after all. Of course I'd be flattered to know that someone is putting it to good use outside EmbedIO, but I think the chances are really thin.

File: src/Swan.Lite/Extensions.IPropertyProxy.cs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        private static string PropertyName<T, TV>(this Expression<Func<T, TV>> propertyExpression)
+        {
+            var memberExpression = !(propertyExpression.Body is MemberExpression)
+                ? (propertyExpression.Body as UnaryExpression).Operand as MemberExpression
+                : propertyExpression.Body as MemberExpression;
+
+            return memberExpression.Member.Name;
+        }
+    }
+}
```
Comment (k3z0) [inline src/Swan.Lite/Extensions.IPropertyProxy.cs]: I suggest changing the file name

**Pull request merged**